### PR TITLE
client: reset a ReqReader if it is necessary to re-authenticate

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -215,8 +217,20 @@ func (c *authenticatingClient) SendRequest(
 	method, svcType, apiVersion, apiCall string,
 	requestData *goosehttp.RequestData,
 ) (err error) {
+	var rs io.ReadSeeker
```
Comment (rogpeppe) [inline client/client.go:220]: If you don't declare err in the return parameters, you won't need this.

File: internal/io/seekable.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the LGPLv3, see COPYING and COPYING.LESSER file for details.
+
+package io
```
Comment (rogpeppe) [inline internal/io/seekable.go]: Could we please not name this the same as the stdlib io package?

File: internal/io/seekable.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	"fmt"
+	"io"
+
+	"gopkg.in/goose.v2/errors"
+)
+
+// maxBufSize holds the maximum amount of data
+// that may be allocated as a buffer when sending
+// a request when the body is not seekable.
+const maxBufSize = 1024 * 1024 * 1024
```
Comment (rogpeppe) [inline internal/io/seekable.go]: This is still quite large... I wonder if we should make buffering scheme pluggable. Something for later.

File: nova/live_test.go [hunk 4]
```diff
@@ -295,7 +295,7 @@ func (s *LiveTests) TestCreateAndDeleteSecurityGroup(c *gc.C) {
 		err = s.nova.DeleteSecurityGroup(group.Id)
 		c.Check(err, gc.IsNil)
 	} else {
-		c.Fatalf("test security group (%d) not found", group.Id)
+		c.Fatalf("test security group (%s) not found", group.Id)
```
Comment (rogpeppe) [inline nova/live_test.go:298]: yay for go vet!

**Review by rogpeppe — APPROVED** [review]
LGTM with a couple of minor suggestions, thanks.

File: client/client.go (hunk 1 above)
Comment (mhilton) [reply client/client.go:220]: I'm leaving this as it was already like that.

File: internal/io/seekable.go (hunk 2 above)
Comment (mhilton) [reply internal/io/seekable.go]: done

**Pull request merged**